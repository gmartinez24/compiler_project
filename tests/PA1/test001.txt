// long test
main

int input, myvar;

function getSumToNIter (int n) : int
{
    int i, j;
    
    i = 0;
    j = 1;
    while (j <= n) do
        i = i + j;
        j++;
    od;
    return i;
};

function getSumToNRecur (int n) : int
{
    if (n == 1) then
        return 1;
    fi;
    return call getSumToNRecur(n - 1) + n;
};

{
    input = call readInt();
    myvar = call getSumToNIter(input);

    call printInt(input);
    call printInt(myvar);
    call println();

    myvar = call getSumToNRecur(input);

    call printInt(input);
    call printInt(myvar);
    call println();
}.
