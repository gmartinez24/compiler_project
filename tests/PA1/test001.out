MAIN
INT
IDENT	input
COMMA
IDENT	myvar
SEMICOLON
FUNC
IDENT	getSumToNIter
OPEN_PAREN
INT
IDENT	n
CLOSE_PAREN
COLON
INT
OPEN_BRACE
INT
IDENT	i
COMMA
IDENT	j
SEMICOLON
IDENT	i
ASSIGN
INT_VAL	0
SEMICOLON
IDENT	j
ASSIGN
INT_VAL	1
SEMICOLON
WHILE
OPEN_PAREN
IDENT	j
LESS_EQUAL
IDENT	n
CLOSE_PAREN
DO
IDENT	i
ASSIGN
IDENT	i
ADD
IDENT	j
SEMICOLON
IDENT	j
UNI_INC
SEMICOLON
OD
SEMICOLON
RETURN
IDENT	i
SEMICOLON
CLOSE_BRACE
SEMICOLON
FUNC
IDENT	getSumToNRecur
OPEN_PAREN
INT
IDENT	n
CLOSE_PAREN
COLON
INT
OPEN_BRACE
IF
OPEN_PAREN
IDENT	n
EQUAL_TO
INT_VAL	1
CLOSE_PAREN
THEN
RETURN
INT_VAL	1
SEMICOLON
FI
SEMICOLON
RETURN
CALL
IDENT	getSumToNRecur
OPEN_PAREN
IDENT	n
SUB
INT_VAL	1
CLOSE_PAREN
ADD
IDENT	n
SEMICOLON
CLOSE_BRACE
SEMICOLON
OPEN_BRACE
IDENT	input
ASSIGN
CALL
IDENT	readInt
OPEN_PAREN
CLOSE_PAREN
SEMICOLON
IDENT	myvar
ASSIGN
CALL
IDENT	getSumToNIter
OPEN_PAREN
IDENT	input
CLOSE_PAREN
SEMICOLON
CALL
IDENT	printInt
OPEN_PAREN
IDENT	input
CLOSE_PAREN
SEMICOLON
CALL
IDENT	printInt
OPEN_PAREN
IDENT	myvar
CLOSE_PAREN
SEMICOLON
CALL
IDENT	println
OPEN_PAREN
CLOSE_PAREN
SEMICOLON
IDENT	myvar
ASSIGN
CALL
IDENT	getSumToNRecur
OPEN_PAREN
IDENT	input
CLOSE_PAREN
SEMICOLON
CALL
IDENT	printInt
OPEN_PAREN
IDENT	input
CLOSE_PAREN
SEMICOLON
CALL
IDENT	printInt
OPEN_PAREN
IDENT	myvar
CLOSE_PAREN
SEMICOLON
CALL
IDENT	println
OPEN_PAREN
CLOSE_PAREN
SEMICOLON
CLOSE_BRACE
PERIOD
EOF
